lib/internal/modules/esm/get_format.js

```diff
@@ -53,7 +55,19 @@ function getFileProtocolModuleFormat(url, context, ignoreErrors) {
   if (experimentalSpecifierResolution !== 'node') {
     // Explicit undefined return indicates load hook should rerun format check
     if (ignoreErrors) return undefined;
-    throw new ERR_UNKNOWN_FILE_EXTENSION(ext, fileURLToPath(url));
+    let suggestion = '';
+    if (getPackageType(url) === 'module' && ext === '') {
+      const config = getPackageScopeConfig(url);
+      const fileBasename = basename(filepath);
+      const relativePath = StringPrototypeSlice(relative(config.pjsonPath, filepath), 1);
+      suggestion = 'Extension-less files are unsupported inside of ' +
```

Comment:
nit
```suggestion
      suggestion = 'Loading extension-less files is not supported inside of ' +
```

Reply:
i usually don't see "extensionless" hyphenated, fwiw

Reply:
@ljharb That is how the esm.md has it: https://github.com/nodejs/node/blame/master/doc/api/esm.md#L1094 , if we want to make it into a terminology/style thing I'd be fine with either but likely not part of this PR. I can't find any reference to the space delimited form in the codebase.